evm/validation.py

```diff
@@ -197,16 +197,11 @@ def validate_unique(values, title="Value"):
         )
 
 
-def validate_block_number(block_number, title="Block Number"):
-    validate_is_integer(block_number, title="Block Number")
-    validate_gte(block_number, 0, title="Block Number")
-
-
 def validate_vm_block_numbers(vm_block_numbers):
     validate_unique(vm_block_numbers, title="Block Number set")
 
     for block_number in vm_block_numbers:
-        validate_block_number(block_number)
+        validate_uint256(block_number)
```

Comment:
Isn't this going the opposite direction of the other type check that says block numbers are more special than just a uint? I'm not sure which one is better, but it seems like consistency is best.

Reply:
I think these are two separate issues that don't actually overlap by much.

- type hints let us say "this integer is a block number, this integer is not a block number, don't allow regular integers in places where we should be using block numbers"
- validation is for "a block number *should* conform to these properties"

Either way this PR is being closed without merging due to figuring out that block numbers are not constrained by any upper bound.